File: smaller-number-than-current.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var smallerNumbersThanCurrent = function(nums) {
  // sort the nums
  // index of the sorted nums
  // compose the new array

  const sortedNums = Array.from(nums).sort((a, b) => a - b);
  return nums.map(num => {
    return sortedNums.indexOf(num);
  });
};

File: smaller-number-than-current.java
class Solution {
    public int[] smallerNumbersThanCurrent(int[] nums) {
        int[] sortedNums = Arrays.copyOf(nums, nums.length);
        Arrays.sort(sortedNums);
        for (int i = 0; i < nums.length; i++) {
            int currNum = nums[i];
            int value = Arrays.stream(sortedNums).boxed().collect(Collectors.toList()).indexOf(currNum);
            nums[i] = value;
        }
        return nums;
    
    }
}